### DISCUSSION TITLE
Feature Request: Add a "metadataObject" property

### DISCUSSION THREAD

**Discussion by nCubed** [keygen-sh/community#36] [Ideas]
Several of the objects use the `metadata` property (product, user, license, machine, policy). The is great, but the downside is a write operation will overwrite all old values, so if one of the existing values is not included in the write operation, it will be deleted.

Would be nice if an additional single `metadataObject` was added as a sibling to the existing `meatadata` object. Where the `metadataObject` is a free-form property we could store our own json or similar. In this way, we would always be retrieving a single object, deserializing, updating, then reserializing back.

I can see how this feature could be abused by extremely large json files, so maybe a size cap would be needed as well.

Thanks.

**Comment by ezekg** [comment]
The [`metadata` attribute](https://keygen.sh/docs/api/metadata/) is already free-form. You can already store JSON in the `metadata` object, so I'm not sure what purpose `metadataObject` would serve here. If you need to merge values, you should merge them before writing.

**Reply by nCubed** [reply]
The metadata value is restricted to 512 characters. The value is having a much larger field to store a larger json object. I fully understand the merging of existing values; this is just a different approach.

**Reply by ezekg** [reply]
We don't plan on supporting larger `metadata` payloads, for performance and security reasons. If you need to store larger JSON payloads, you should store it outside of Keygen and perform a lookup based on e.g. license ID.

You will need to work within the constraints of the current attribute.
